globus_cli/commands/endpoint/update.py

```diff
@@ -11,23 +13,35 @@
 @common_options
 @endpoint_id_arg
 @endpoint_create_and_update_params(create=False)
-def endpoint_update(endpoint_id, display_name, description, organization,
-                    department, keywords, contact_email, contact_info,
-                    info_link, public, default_directory, force_encryption,
-                    oauth_server, myproxy_server, myproxy_dn):
+def endpoint_update(endpoint_id, display_name, description, info_link,
+                    contact_info, contact_email, organization, department,
+                    keywords, public, location, disable_verify, myproxy_dn,
+                    myproxy_server, oauth_server, force_encryption,
+                    default_directory, subscription_id, network_use,
+                    max_concurrency, preferred_concurrency,
+                    max_parallelism, preferred_parallelism):
     """
     Executor for `globus endpoint update`
     """
+    # validate params. Requires a get call to check the endpoint type
+    args, _, _, values = inspect.getargvalues(inspect.currentframe())
+    params = dict((k, values[k]) for k in args)
```

Comment:
I'm curious why you're using `inspect` for this purpose. It seems to me much cleaner to take the keyword args dict, `**kwargs`, for all of the arguments above.

`kwargs` then replaces all usage of `params` below.